### PR TITLE
ban the usage of Endpoint type

### LINKED ISSUE #92: disallow the usage of legacy Endpoint type similar to FullResponse
Endpoint along with FullResponse are the legacy typings built around fixture/serve-runtime service wrappers, later on we created newer and more specific service API typings. eventually they will all be gone once everything is migrated to use `fetch` api.
To report error of the usage of such legacy typings will make the transition smooth.

### PR DESCRIPTION
**Pull request by le-cong**
Fixes #92

### CODE REVIEW

**Review by carlansley — APPROVED** [review]

**Comment by le-cong** [comment]
sorry, can you re-approve due to conflict merge?

File: package.json [hunk 1]
```diff
@@ -63,13 +63,15 @@
   "dependencies": {
     "@typescript-eslint/type-utils": "^8.15.0",
     "@typescript-eslint/utils": "^8.15.0",
+    "debug": "^4.3.7",
```
Comment (carlansley) [inline package.json:66]: i don't see this being used anywhere

File: package.json [hunk 2]
```diff
@@ -63,13 +63,15 @@
...
     "ts-api-utils": "^1.4.0"
   },
   "devDependencies": {
     "@checkdigit/jest-config": "^6.0.2",
     "@checkdigit/prettier-config": "^5.5.1",
     "@checkdigit/typescript-config": "^8.0.0",
     "@eslint/js": "^9.15.0",
+    "@types/debug": "^4.1.12",
```
Comment (carlansley) [inline package.json:74]: this also

**Review by carlansley — CHANGES_REQUESTED** [review]
lftm, just a dep thing

**Review by carlansley — APPROVED** [review]

**Pull request merged**